strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactMetadataServiceSnapshotsTest.java

```diff
@@ -177,56 +160,44 @@ public void testAddTimestampedSnapshotVersionToMetadata()
-                                                                          "deleted",
-                                                                          "2.0",
-                                                                          "jar",
-                                                                          CLASSIFIERS,
-                                                                          5);
-
-        changeCreationDate(deletedArtifact);
-
         String artifactPath = "org/carlspring/strongbox/deleted";
 
         artifactMetadataService.rebuildMetadata(STORAGE0, REPOSITORY_SNAPSHOTS, artifactPath);
 
-        String metadataPath = artifactPath + "/2.0-SNAPSHOT";
+        String metadataPath = artifactPath + "/1.0-SNAPSHOT";
 
         Metadata metadataBefore = artifactMetadataService.getMetadata(STORAGE0, REPOSITORY_SNAPSHOTS, metadataPath);
+        for (Path path : snapshotArtifacts)
+        {
+            MavenArtifactCoordinates coordinates = (MavenArtifactCoordinates) RepositoryFiles.readCoordinates((RepositoryPath)path.normalize());
+            assertTrue(MetadataHelper.containsTimestampedSnapshotVersion(metadataBefore, coordinates.getVersion(), null));
+        }
 
-        assertTrue(
-                MetadataHelper.containsTimestampedSnapshotVersion(metadataBefore, deletedArtifact.getVersion(), null));
-
+        MavenArtifactCoordinates coordinates = (MavenArtifactCoordinates) RepositoryFiles.readCoordinates((RepositoryPath)snapshotArtifacts.iterator().next().normalize());
         artifactMetadataService.removeTimestampedSnapshotVersion(STORAGE0,
                                                                  REPOSITORY_SNAPSHOTS,
                                                                  artifactPath,
-                                                                 deletedArtifact.getVersion(),
+                                                                 coordinates.getVersion(),
                                                                  null);
 
         Metadata metadataAfter = artifactMetadataService.getMetadata(STORAGE0, REPOSITORY_SNAPSHOTS, metadataPath);
-
         assertNotNull(metadataAfter);
-        assertFalse(MetadataHelper.containsTimestampedSnapshotVersion(metadataAfter, deletedArtifact.getVersion()));
+        assertFalse(MetadataHelper.containsTimestampedSnapshotVersion(metadataAfter, coordinates.getVersion()));
     }
 
     @Test
-    public void testSnapshotWithoutTimestampMetadataRebuild()
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class, ArtifactManagementTestExecutionListener.class })
+    public void testSnapshotWithoutTimestampMetadataRebuild(@MavenSnapshotRepository Repository repository,
+                                                            @MavenArtifactWithClassifiers(repository = REPOSITORY_SNAPSHOTS, id = "org.carlspring.strongbox.snapshots:metadata", versions = "2.0-SNAPSHOT") List<Path> snapshotArtifactPath)
```

Comment:
So, how do you actually specify what classifiers to use when generating the artifacts? I'm not sure I get it.

Reply:
Classifiers defined in `MavenArtifactWithClassifiersGeneratorStrategy.CLASSIFIERS`

Reply:
@carlspring classifiers parameter implemented as you requested (see [here](https://github.com/strongbox/strongbox/pull/1199/files#diff-d58def3a919b56e55a36134fc5c5e122R313))

Reply:
Thank you very much! :)